bin/lint_go

```diff
@@ -6,6 +6,11 @@ if [ -n "$(goimports -d src 2>&1)" ]; then
   exit 1
 fi
 
-# Disabled linters:
-# - https://github.com/GoASTScanner/gas: not production-ready at this point
-gometalinter.v1 -D gas src/...
+gometalinter --debug src/... 2> lint_output.txt
+if [ -n "$(grep 'warning:' lint_output.txt)" ]; then
+  echo "There are linter warnings. Check 'lint_output.txt'."
+  echo ""
+  echo "Expected warnings:"
+  echo "gocyclo - deserializeStep (ignored with a nolint directive)"
+  echo "gotype - see https://github.com/alecthomas/gometalinter/issues/348"
+fi
```

Comment:
Cat the lint_output file and delete it here, so that we don't have to add it to gitignore?

Reply:
Why not simply display the linter output here and stop the script it it has an exit code >0?

Reply:
It already exits kif the exit code is greater than 0. gometalinter issues warnings anytime a linter exits with status 0. Some of these are normal as a linter may report errors that you ignore through gometalinter. Some are not normal though and cause some linters to run fully and thus not report lint errors. We ran into this on exosphere.

Reply:
Its a lot of output. I didn't want to parse it as its debug output (and thus i think could change just about anytime) and would be annoying to parse (as grepping for "warning" doesn't give enough context.